hack/install-go.sh

```diff
@@ -1,8 +1,28 @@
 #!/bin/bash -xe
 
 destination=$1
-version=$(grep "^go " go.mod |awk '{print $2}')
-tarball=go$version.linux-amd64.tar.gz
+
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+go_mod_path="$script_dir/../go.mod"
+
+go_mod_version=$(grep '^toolchain' "$go_mod_path" 2>/dev/null | awk '{print $2}' | sed 's/^go//' || true)
+if [ -z "$go_mod_version" ]; then
+    go_mod_version=$(grep '^go ' "$go_mod_path" | awk '{print $2}')
+fi
+
+if [[ "$go_mod_version" =~ ^[0-9]+\.[0-9]+$ ]]; then
+    minor_version=$(echo "$go_mod_version" | cut -d. -f2)
+    # Starting from Go 1.21, the initial release is 1.N.0
+    # For Go 1.20 and earlier, the initial release is 1.N
+    # For more information, see https://go.dev/doc/toolchain#name
+    if [ "$minor_version" -ge 21 ]; then
+        go_mod_version="${go_mod_version}.0"
+    fi
```

Comment:
on this release branch lets do the minimal required, without all the logic, since it just need 1.N

unless you plan something generic that you can put on main and backport, but if we can refrain from it, and keep it simple it would be better please

it seems the logic adds and then remove the `.z`

Reply:
yep